app/models/vendor.rb

```diff
@@ -73,4 +73,17 @@ def status
       product_counts.key?(product_state) ? product_counts[product_state].count : 0
     end
   end
+
+  def header_fields?
+    url? || address? || !points_of_contact.empty?
+  end
+
+  def product_arr
+    Product.where(vendor_id: id).by_updated_at.order_by(state: 'desc')
```

Comment:
There is already an association between vendors and product, it makes sense here to just say `products.by_updated_at.order_by(state: 'desc')`. Also this whole method seems like it could be replaced with a scope on the product called `ordered_for_vendors`.